### PR TITLE
Ret 1725

### PR DESCRIPTION
**Pull request by lvo700**
### JIRA link

https://tools.hmcts.net/jira/browse/RET-1725

### Change description

Validation and error messages for Employment Details (Working) route.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by marcjudge — COMMENTED** [review]
Sonar Cloud is highlighting a few code smells. Please try to resolve these
https://sonarcloud.io/summary/new_code?id=et-sya&pullRequest=240

**Comment by lvo700** [comment]
> Sonar Cloud is highlighting a few code smells. Please try to resolve these https://sonarcloud.io/summary/new_code?id=et-sya&pullRequest=240

Fixed sonarcloud issues

**Comment by marcjudge** [comment]
> > Sonar Cloud is highlighting a few code smells. Please try to resolve these https://sonarcloud.io/summary/new_code?id=et-sya&pullRequest=240
> 
> Fixed sonarcloud issues

Could you also please increase the code coverage here - https://sonarcloud.io/component_measures?metric=new_coverage&selected=et-sya%3Asrc%2Fmain%2Fcontrollers%2Fhelpers.ts&view=list&pullRequest=240&id=et-sya? A few scenarios around the type of errors

**Comment by lvo700** [comment]
> > > Sonar Cloud is highlighting a few code smells. Please try to resolve these https://sonarcloud.io/summary/new_code?id=et-sya&pullRequest=240
> > 
> > 
> > Fixed sonarcloud issues
> 
> Could you also please increase the code coverage here - https://sonarcloud.io/component_measures?metric=new_coverage&selected=et-sya%3Asrc%2Fmain%2Fcontrollers%2Fhelpers.ts&view=list&pullRequest=240&id=et-sya? A few scenarios around the type of errors

Increased coverage

File: src/main/auth/index.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const getUserDetails = async (
   const tokenUrl: string = config.get('services.idam.tokenURL');
   const callbackUrl = encodeURI(serviceUrl + callbackUrlPageLink);
   const code = encodeURIComponent(rawCode);
-  const data = `client_id=${id}&client_secret=${secret}&grant_type=authorization_code&redirect_uri=${callbackUrl}&code=${code}`;
+  const data = `client_id=${id}&client_secret=${secret}&grant_type=authorization_code&redirect_uri=${callbackUrl}&code=${code}&scope=roles&username=et.dev@hmcts.net&password=Pa55word11`;
```
Comment (jsinghm13) [inline src/main/auth/index.ts:25]: You need to revert this change, it is for local testing only

**Review by marcjudge — APPROVED** [review]

File: src/main/auth/index.ts (hunk 1 above)
Comment (lvo700) [reply src/main/auth/index.ts]: Reverted

File: src/main/components/form/validator.ts [hunk 2]
```diff
@@ -189,10 +189,12 @@ export const isValidInteger: Validator = value => {
...
 
-  if (/^\d+$/.test(value as string)) {
-    return;
-  } else {
-    return 'invalid';
+  if (/^\D+$/.test(value as string) || /^\d+\D+$/.test(value as string)) {
+    return 'notANumber';
+  }
+
+  if (/^\D+$/.test(value as string) || /^\d+[^0-9.]+$/.test(value as string)) {
```
Comment (jsinghm13) [inline src/main/components/form/validator.ts:196]: remove first condition and move the second condition to above if statement

Comment (lvo700) [reply src/main/components/form/validator.ts:192]: Fixed duplicate

File: src/test/unit/components/validator.test.ts [hunk 3]
```diff
@@ -382,10 +420,10 @@ describe('Validation', () => {
 
   describe('isValidCurrency()', () => {
     it.each([
-      { mockRef: '', expected: undefined },
```
[the remark this answers is not in the document]
Comment (lvo700) [reply src/test/unit/components/validator.test.ts:385]: Added checks

File: src/test/unit/components/validator.test.ts [hunk 4]
```diff
@@ -233,9 +235,12 @@ describe('Validation', () => {
...
       { mockRef: null, expected: 'invalid' },
-      { mockRef: 'a', expected: 'invalid' },
-      { mockRef: '%', expected: 'invalid' },
-      { mockRef: '25a', expected: 'invalid' },
+      { mockRef: 'a', expected: 'notANumber' },
+      { mockRef: '%', expected: 'notANumber' },
+      { mockRef: '25a', expected: 'notANumber' },
+      { mockRef: 'a', expected: 'notANumber' },
+      { mockRef: '%', expected: 'notANumber' },
+      { mockRef: '25a', expected: 'notANumber' },
```
Comment (jsinghm13) [inline src/test/unit/components/validator.test.ts:243]: either you intended to test different values here or accidentally duplicated, please fix

Comment (lvo700) [reply src/test/unit/components/validator.test.ts]: Removed duplicates

**Review by jsinghm13 — APPROVED** [review]
